features/step_definitions/sample_file_steps.rb

```diff
@@ -1,3 +1,22 @@
+Given(/^the smelly file 'smelly.rb'$/) do
+  contents = <<-EOS.strip_heredoc
+    class Smelly
+      def x; end
+    end
+  EOS
+  write_file('smelly.rb', contents)
+end
+
+Given(/^the clean file 'clean.rb'$/) do
+  contents = <<-EOS.strip_heredoc
+    # Explanatory comment
+    class Clean
+      def foo; end
+    end
+  EOS
+  write_file('clean.rb', contents)
+end
```

Comment:
I believe we could and should reduce and unify all of our used code samples since they are quite a few that overlap. But that would be a separate pull request.